src/coprocessor/Client.cc

```diff
@@ -123,19 +123,36 @@ std::vector<BatchCopTask> balanceBatchCopTasks(std::vector<BatchCopTask> && orig
     }
 
     // Only one tiflash store
-    if (original_tasks.size() <= 1)
+    if (original_tasks.size() <= 1 && !is_mpp)
     {
         return std::move(original_tasks);
     }
 
     std::map<uint64_t, BatchCopTask> store_task_map;
-    for (const auto & task : original_tasks)
+    if (!is_mpp)
```

Comment:
just like: https://github.com/pingcap/tidb/blob/master/store/copr/batch_coprocessor.go#L318

Reply:
Otherwise if there is only one batchCopTask in `original_tasks`, we cannot balance.